### PR TITLE
Remove RANDFILE settings from configuration files

### LINKED ISSUE #10433: Remove .rnd use from tests, config, etc.
There are many instances of .rnd file  Most of this can and should be removed:
```
;g grep --fixed .rnd
.gitignore:/ms/.rnd
CHANGES:  *) In versions up to 0.9.6, RAND_file_name() resorted to file ".rnd"
crypto/rand/randfile.c:#define RFILE ".rnd"
demos/certs/apps/apps.cnf:RANDFILE              = $ENV::HOME/.rnd
demos/certs/ca.cnf:RANDFILE             = $ENV::HOME/.rnd
doc/man1/openssl-ca.pod.in: ./demoCA/.rnd                  - CA random seed information
doc/man1/openssl-req.pod.in: RANDFILE               = $ENV::HOME/.rnd
doc/man3/RAND_load_file.pod:Otherwise, the file is called C<.rnd>, found in platform dependent locations:
doc/man5/config.pod: RANDFILE= ${ENV::HOME}/.rnd
test/CAss.cnf:RANDFILE          = ./.rnd
test/CAssdh.cnf:RANDFILE              = ./.rnd
test/CAssdsa.cnf:RANDFILE              = ./.rnd
test/CAssrsa.cnf:RANDFILE              = ./.rnd
test/CAtsa.cnf:RANDFILE         = ./.rnd
test/P1ss.cnf:RANDFILE          = ./.rnd
test/P2ss.cnf:RANDFILE          = ./.rnd
test/Sssdsa.cnf:RANDFILE              = ./.rnd
test/Sssrsa.cnf:RANDFILE              = ./.rnd
test/Uss.cnf:RANDFILE           = ./.rnd
test/conf_include_test.c:    if (!TEST_ptr(str) || !TEST_str_eq(str, "./.rnd")) {
test/recipes/25-test_req.t:open RND, ">>", ".rnd";
test/recipes/80-test_ssl_old.t:    open RND, ">>", ".rnd";
test/recipes/90-test_includes_data/conf-includes/includes1.cnf:RANDFILE         = ./.rnd
test/smime-certs/ca.cnf:RANDFILE                = $ENV::HOME/.rnd
test/test.cnf:RANDFILE          = ./.rnd
```

### PR DESCRIPTION
**Pull request by mspncp**
Fixes #10433

_(edited, because pull request was created prematurely before I was finished)_

This is marked WIP, because up to now I only removed the tests.

I plan to also update the demos and the manual pages.

##### Checklist

- [x] tests are added or updated
- [x] demos are added or updated
- [x] documentation is added or updated

### CODE REVIEW

**Comment by mspncp** [comment]
_(Force-pushed to add `[extended tests]` to commit message)_

**Comment by levitte** [comment]
This didn't help the Appveyor failure at least.  I've seen this odd thing a few times now:

```
unable to get '$ENV::TSDNSECT' section
```

**Comment by richsalz** [comment]
```
; g grep TSDNSECT
test/CAtsa.cnf:TSDNSECT         = ts_cert_dn
test/CAtsa.cnf:distinguished_name       = $ENV::TSDNSECT
test/recipes/80-test_tsa.t:    $ENV{TSDNSECT} = "ts_cert_dn";
test/recipes/80-test_tsa.t:     $ENV{TSDNSECT} = "ts_ca_dn";
```

**Comment by richsalz** [comment]
Is it possible perl on appvery doesn't handle ```$ENV``` right?

**Comment by mattcaswell** [comment]
Looks good so far

**Comment by levitte** [comment]
Ah!  I think the `$ENV::TSDNSECT` will be resolved by #10428

**Comment by mspncp** [comment]
Rebased without changes (before continuing wip) to pick up the fix from #10428.

**Comment by mspncp** [comment]
Completed the pull request and squashed everything with a new commit message. Taking it out of WIP.

```
commit 174cba8f87e08cf2324204ed3390ec156d314f77
Author: Dr. Matthias St. Pierre <Matthias.St.Pierre@ncp-e.com>
Date:   Wed Nov 13 16:02:09 2019 +0100

    Remove RANDFILE settings from configuration files
    
    OpenSSL 1.1.1 introduced a new CSPRNG with an improved seeding
    mechanism, which makes it dispensable to define a RANDFILE for
    saving and restoring randomness. This commit removes the RANDFILE
    declarations from our own configuration files and adds documentation
    that this option is not needed anymore and retained mainly for
    compatibility reasons.
    
    [extended tests]
    
    Fixes #10433
```

File: doc/man1/openssl-ca.pod.in [hunk 1]
```diff
@@ -442,7 +442,8 @@ CA private key. Mandatory.
 =item B<RANDFILE>
 
 At startup the specified file is loaded into the random number generator,
-and at exit 256 bytes will be written to it.
+and at exit 256 bytes will be written to it. (Note: Using a RANDFILE is
+not necessary anymore, see the L</HISTORY> section.
```
Comment (mspncp) [inline doc/man1/openssl-ca.pod.in]: @levitte is this the correct way to reference sections in pod files? In other places, I see a `B<..>` around the reference:

https://github.com/openssl/openssl/blob/174cba8f87e08cf2324204ed3390ec156d314f77/doc/man1/openssl-ca.pod.in#L112-L113

Comment (levitte) [reply doc/man1/openssl-ca.pod.in]: Yes, that's the correct way to do it (I assume you mean within the same POD file).
The references done with B\<...\> are incorrect (that's technical depth, we were not as aware of POD markup back a number of years)

File: doc/man1/openssl-ca.pod.in [hunk 2]
```diff
@@ -763,6 +762,11 @@ B<-enddate> and B<-days>) will be encoded as UTCTime if the dates are
 earlier than year 2049 (included), and as GeneralizedTime if the dates
 are in year 2050 or later.
 
+OpenSSL 1.1.1 introduced a new random generator (CSPRNG) with an improved
+seeding mechanism. The new seeding mechanism makes it dispensable to
```
Comment (paulidale) [inline doc/man1/openssl-ca.pod.in]: I'd suggest: `s/dispensable/unnecessary/`.
Dispensable seems odd here.

File: doc/man1/openssl-ts.pod.in [hunk 3]
```diff
@@ -644,6 +652,13 @@ test/testtsa).
 
 =back
 
+=head1 HISTORY
+
+OpenSSL 1.1.1 introduced a new random generator (CSPRNG) with an improved
+seeding mechanism. The new seeding mechanism makes it dispensable to
```
Comment (paulidale) [inline doc/man1/openssl-ts.pod.in]: Same here.

File: doc/man1/openssl-ca.pod.in (hunk 2 above)
Comment (mspncp) [reply doc/man1/openssl-ca.pod.in]: Thanks for the hint, I'll fix it. Such things happen when one just looks up a term in the dictionary (here: [verzichtbar](https://www.dict.cc/?s=verzichtbar)) and chooses from a list of possible translations.

Comment (mspncp) [reply doc/man1/openssl-ca.pod.in]: Done in a4d887541fe2d28c183237274160b9e419f8b7a9.

Comment (paulidale) [reply doc/man1/openssl-ca.pod.in]: I suspected it was something like that.  English has a large number of similar meaning words each with their own undertones and nuances.

**Comment by mspncp** [comment]
Merged to master in 23f3993127c0a05651e28701d91edb478ebe6efa, thanks!

**Pull request closed without merging**